src/components/grid-list/tile-styler.ts

```diff
@@ -0,0 +1,194 @@
+    tile.setStyle('top', this.getTilePosition(this.fixedRowHeight, rowIndex));
+    tile.setStyle('height', calc(this.getTileSize(this.fixedRowHeight, tile.rowspan)));
+  }
+
+  getComputedHeight(): [string, string] {
+    return ['height', calc(`${this._rowspan} * ${this.getTileSize(this.fixedRowHeight, 1)}`)];
+  }
+}
+
+/*  This type of styler is instantiated when the user passes in a width:height ratio
+ *  for the row height.  Example <md-grid-list cols="3" rowHeight="3:1"> */
+export class RatioTileStyler extends TileStyler {
+
+  /** Ratio width:height given by user to determine row height.*/
+  rowHeightRatio: number;
+  baseTileHeight: string;
+
+  constructor(value: string) {
+    super();
+    this._parseRatio(value);
+  }
+
+  setRowStyles(tile: MdGridTile, rowIndex: number, percentWidth: number,
+               gutterWidth: number): void {
+    let percentHeightPerTile = percentWidth / this.rowHeightRatio;
+    this.baseTileHeight = this.getBaseTileSize(percentHeightPerTile, gutterWidth);
+
+    // Use paddingTop and marginTop to maintain the given aspect ratio, as
+    // a percentage-based value for these properties is applied versus the *width* of the
+    // containing block. See http://www.w3.org/TR/CSS2/box.html#margin-properties
+    tile.setStyle('marginTop', this.getTilePosition(this.baseTileHeight, rowIndex));
+    tile.setStyle('paddingTop', calc(this.getTileSize(this.baseTileHeight, tile.rowspan)));
+  }
+
+  getComputedHeight(): [string, string] {
+    return [
+      'paddingBottom', calc(`${this._rowspan} * ${this.getTileSize(this.baseTileHeight, 1)}`)
+    ];
+  }
+
+  _parseRatio(value: string): void {
+    let ratioParts = value.split(':');
+
+    if (ratioParts.length !== 2) {
+      throw new MdGridListBadRatioError(value);
+    }
+
+    this.rowHeightRatio = parseFloat(ratioParts[0]) / parseFloat(ratioParts[1]);
+  }
+}
+
+/*  This type of styler is instantiated when the user selects a "fit" row height mode.
+ *  In other words, the row height will reflect the total height of the container divided
+ *  by the number of rows.  Example <md-grid-list cols="3" rowHeight="fit"> */
+export class FitTileStyler extends TileStyler {
+
+  setRowStyles(tile: MdGridTile, rowIndex: number, percentWidth: number,
+               gutterWidth: number): void {
+    // Percent of the available vertical space that one row takes up.
+    let percentHeightPerTile = 100 / this._rowspan;
+
+    // Fraction of the horizontal gutter size that each column takes up.
+    let gutterHeightPerTile = (this._rows - 1) / this._rows;
+
+    // Base vertical size of a column.
+    let baseTileHeight = this.getBaseTileSize(percentHeightPerTile, gutterHeightPerTile);
+
+    tile.setStyle('top', this.getTilePosition(baseTileHeight, rowIndex));
+    tile.setStyle('height', calc(this.getTileSize(baseTileHeight, tile.rowspan)));
+  }
+}
+
+/** Wraps a CSS string in a calc function */
+function calc(exp: string): string { return `calc(${exp})`; }
```

Comment:
Using this function is the same amount of characters as not using it. Would it make more sense to just inline it everywhere?